docs/fsharp/language-reference/slices.md

```diff
@@ -143,6 +143,64 @@ let xs = [1 .. 10]
 printfn "%A" xs.[2..5] // Includes the 5th index
 ```
 
+## Built-in F# slices produce empty slices for slices that are impossible to generate
+
+F# lists, arrays, sequences, strings, 2D arrays, 3D arrays, and 4D arrays will all produce an empty slice if the syntax could produce a slice that doesn't exist.
+
+Consider the following:
+
+```fsharp
+let l = [ 1..10 ]
+let a = [| 1..10 |]
+let s = "hello!"
+
+let emptyList = l.[-2..(-1)]
+let emptyArray = a.[-2..(-1)]
+let emptyString = s.[-2..(-1)]
+```
+
+C# developers may expect these to throw an exception rather than produce an empty slice. This is a design decision rooted in the fact that empty collections compose in F#. An empty F# list can be composed with another F# list, an empty string can be added to an existing string, and so on. It can be common to take slices based on values passed in as parameters, and being tolerant of out-of-bounds by producing an empty collection fits with the composition nature of F# code.
+
+## Fixed-index slices for 3D and 4D arrays
+
+For F# 3D and 4D arrays, you can "fix" a particular index and slice other dimensions with that index fixed.
+
+To illustrate this, consider the following 3D array:
+
+*z = 0*
+|x\y|0|1|
+|---|-|-|
+|**0**|0|1|
+|**1**|2|3|
+
+*z = 1*
+|x\y|0|1|
+|---|-|-|
+|**0**|4|5|
+|**1**|6|7|
```

Comment:
```suggestion
*z = 0*
| x\y   | 0 | 1 |
|-------|---|---|
| **0** | 0 | 1 |
| **1** | 2 | 3 |

*z = 1*
| x\y   | 0 | 1 |
|-------|---|---|
| **0** | 4 | 5 |
| **1** | 6 | 7 |
```

Fix formatting so that the table is render correctly.